src/microbe_stage/Membrane.cs

```diff
@@ -201,6 +210,32 @@ public bool Contains(float x, float y)
         return crosses;
     }
 
+    public void EnableEngulfAnimation(bool enable, double delta)
+    {
+        if (enable && engulfFade < 1)
+        {
+            engulfFade += delta / 0.5;
+            engulfFade = Math.Min(engulfFade, 1);
+        }
+        else if (!enable && engulfFade > 0)
+        {
+            engulfFade -= delta / 0.5;
+            engulfFade = Math.Max(engulfFade, 0);
+        }
+
+        if (engulfFade != 0)
+        {
+            EngulfShaderMaterial?.SetShaderParameter(fadeParameterName, engulfFade);
```

Comment:
I think the parameter to the shader here should be sent as a float as shaders don't use doubles (usually for performance reasons).